src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -645,6 +645,44 @@ namespace smt::noodler {
         return true;
     }
 
+    bool DecisionProcedure::check_diseq(const AutAssignment& ass, const std::pair<BasicTerm, BasicTerm>& pr) {
+        auto get_symbol_word = [](Nfa &nfa) { // assumes that nfa accepts words of size 0 or 1
+            nfa.trim();
+            std::set<Mata::Symbol> symbols;
+            for (const auto &tran : nfa.delta) {
+                symbols.insert(tran.symb);
+            }
+            return symbols;
+        };
+
+        auto s1 = get_symbol_word(*ass.at(pr.first));
+        auto s2 = get_symbol_word(*ass.at(pr.second));
+        if(s1.size() == 1 && s2.size() == 1 && s1 == s2) {
```

Comment:
Problem might be in this condition, I'll change it

Reply:
Hmm, it is harder than I thought, we should talk about it. The problem is what to do if `s1` or `s2` is empty, i.e. variable $a_1$ or $a_2$ is only $\epsilon$.

Reply:
If s1 or s2 is empty, the result should be unsat, no?

Reply:
I guess, I am trying it now what happens

Reply:
If language of some variable is empty, I think it should die before we get to this function of checking diseqs

Reply:
The problem is that we do not add $a_1$ and $a_2$ to length variables (created from disequality $L \neq R$, $L = x_1a_1y_1$, $R= x_2a_2y_2$). We need the exact languages which can be only done by splitting, otherwise we have overapproximation of languages of $a_1$ and $a_2$, so checking if they contain some different symbols is not enough. Here the different symbols were the dummy ones, after making them length variables, it gives correct result.

Reply:
It makes sense